sdk/spring/spring-cloud-azure-feature-management/src/test/java/com/azure/spring/cloud/feature/management/ValidationsTest.java

```diff
@@ -0,0 +1,152 @@
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.context.ApplicationContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+public class ValidationsTest {
+    @Mock
+    private ApplicationContext context;
+    @Mock
+    private FeatureManagementConfigProperties configProperties;
+
+    private final ObjectMapper objectMapper = JsonMapper.builder()
+        .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true).build();
+    private final String testCaseFolderPath = "validations-tests";
+    private final String inputsUser = "user";
+    private final String inputsGroups = "groups";
+    private final String sampleFileNameFilter = "sample";
+    private final String testsFileNameFilter = "tests";
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+        when(configProperties.isFailFast()).thenReturn(true);
+        when(context.getBean(Mockito.contains("TimeWindow"))).thenReturn(new TimeWindowFilter());
+    }
+
+    @AfterEach
+    public void cleanup() throws Exception {
+        MockitoAnnotations.openMocks(this).close();
+    }
+
+    private boolean hasException(ValidationTestCase testCase) {
+        final String exceptionStr = testCase.getIsEnabled().getException();
+        return exceptionStr != null && !exceptionStr.isEmpty();
+    }
+
+    private boolean hasInput(ValidationTestCase testCase) {
+        final LinkedHashMap<String, Object> inputsMap = testCase.getInputs();
+        return inputsMap != null && !inputsMap.isEmpty();
+    }
+
+    private File[] getFileList(String fileNameFilter) {
+        final URL folderUrl = Thread.currentThread().getContextClassLoader().getResource(testCaseFolderPath);
+        assert folderUrl != null;
+
+        final File folderFile = new File(folderUrl.getFile());
+        final File[] filteredFiles = folderFile.listFiles(pathname -> pathname.getName().toLowerCase().contains(fileNameFilter));
+        assert filteredFiles != null;
+
+        Arrays.sort(filteredFiles, Comparator.comparing(File::getName));
+        return filteredFiles;
+    }
+
+    private List<ValidationTestCase> readTestcasesFromFile(File testFile) throws IOException {
+        final String jsonString = Files.readString(testFile.toPath());
+        final CollectionType typeReference =
+            TypeFactory.defaultInstance().constructCollectionType(List.class, ValidationTestCase.class);
+        return objectMapper.readValue(jsonString, typeReference);
```

Comment:
Is there a reason we can't just use json instead of creating POJOs. It seems an unneeded step. And will currently break if any new fields are added.